### ISSUE TITLE
make Gnocchi logs accessible

### ISSUE THREAD

**Issue by chungg** [jd/pifpaf#56]
i may be wrong, but it's impossible to debug when using pifpaf (at least for gnocchi). i've no idea what is happening internally since i only really have access to API only.

**Comment by jd** [comment]
You can pass the `--debug` flag to pifpaf so it prints program output debug to stdout. Is there anything specific that you are trying to debug that is not covered by this option?

**Comment by chungg** [comment]
let me try again, when i did --debug all that was printed was the startup information of the driver. i couldn't see metricd.logs.

trying to see what is happening internally. ie for example, if gnocchi is really backed up or processing very slow, from API i can only tell i don't see expected measures but not real issue behind it.

**Comment by jd** [comment]
Right, that's a much better explanation of your problem. No, currently there's no way to have logs because they are not configured. But it should be pretty easy to change the Gnocchi driver to make Gnocchi write log files (when `--debug` is used) in pifpaf's temporary directory. Let me give it a try.

**Comment by jd** [comment]
@chungg Can you give a try to the branch above to see if that helps? This passes `debug = true` to Gnocchi when `pifpaf --debug` is used, the log should be available on stderr, not only the startup one then.

**Comment by chungg** [comment]
got following (i might have edited env wrong):

ERROR: pifpaf: Traceback (most recent call last):
  File "/opt/stack/gnocchi/.tox/py27-mysql-file/lib/python2.7/site-packages/fixtures/fixture.py", line 197, in setUp
    self._setUp()
  File "/opt/stack/gnocchi/.tox/py27-mysql-file/lib/python2.7/site-packages/pifpaf/drivers/gnocchi.py", line 154, in _setUp
    url = %s""" % (self.debug,
AttributeError: 'GnocchiDriver' object has no attribute 'debug'

**Comment by jd** [comment]
Weird, works here. Did you checkout the entire branch at https://github.com/jd/pifpaf/tree/jd/gnocchi-log ? there's 2 commits and the first one adds `self.debug` to drivers.

**Comment by chungg** [comment]
yeah, i missed the first commit. i just gave it a shot and it gives gnocchi metricd output. one issue is that it seems to be only giving logs from the initial startup of service. when running in tox against tests, it doesn't actually show what is happening when tests are running (unless this is being written to somewhere other than stdout)

**Comment by jd** [comment]
If you type `pifpaf --debug run gnocchi $SHELL` you'll see the constant output from metricd on stderr, which tends to indicate that it works. There might be something in tox eating the output?
